src/machine.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  Machine,
+  assign,
+  send,
+} from 'xstate';
+import { syncToLatestBlock } from './utils';
+import {
+  SyncSchema,
+  SyncContext,
+} from './types';
+
+export const syncHandler = (_context, _event) => (callback, onReceive) => {
+  console.log('Sync handler instantiated');
+  const iterator = syncToLatestBlock();
+  const asyncCall: () => void = async () => {
+    for (;;) {
+      const block = await iterator.next();
+
+      const { value, done } = block;
+      if (done) {
+        console.log('Done!', value)
+        break;
+      }
+
+      if (value && !value.success) {
+        console.log('Erroed!', value.message);
+        callback('ERROR');
+        return;
+      }
+
+      if (value.type === 'reorg') {
+        console.log(value.message);
+        callback('REORG');
+        return;
+      }
+
+      if (value.type === 'finished') {
+        console.log('FINISHED!');
+        callback('DONE');
+      }
+
+      console.log(value);
+    }
+
+    return;
+  };
+
+
+  onReceive((e) => {
```

Comment:
Missing docstring and the type of `e`.

Reply:
Done in 6d804938dc8037d93db33c734dc9c13de2c981fc